src/components/TodoView.js

```diff
@@ -0,0 +1,102 @@
+import PropTypes from "prop-types";
+import React from "react";
+import AddTodoForm from "./AddTodoForm";
+import TodoList from "./TodoList";
+import styles from "./SharedStyles.module.css";
+import { FaSortAlphaDown, FaSortAlphaUpAlt } from "react-icons/fa";
+import Toggle from "./Toggle";
+import Footer from "./Footer";
+import GoBack from "./GoBack";
+
+const TodoView = ({
+  onAddTodo,
+  fetchData,
+  updateData,
+  removeTodo,
+  isLoading,
+  todoList,
+  sortOrder,
+  setSortOrder,
+  sortData,
+  isDarkMode,
+  handleSwitch,
+}) => {
+  return (
+    <>
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+        <div
+          className={`${styles.HomePage} ${
+            isDarkMode ? styles.DarkModeHomePage : styles.LightModeHomePage
+          }`}
+        > <div className={styles.topButtonContainer}>
+          <GoBack />
+          <Toggle
+            className={`${
+              isDarkMode
+                ? styles.darkThemeModeButton
+                : styles.lightThemeModeButton
+            }`}
+            isDarkMode={isDarkMode}
+            onSwitch={handleSwitch}
+          />
+          </div>
+          <div className={styles.Text}>
+            <h1>Todo List</h1>
+            <AddTodoForm onAddTodo={onAddTodo} isDarkMode={isDarkMode} />
+            <div className={styles.SortButtonContainer}>
+              <button
+                className={styles.SortButton}
+                onClick={() => {
+                  setSortOrder("asc");
+                  sortData();
+                }}
```

Comment:
Requires change:
Instead of updating state here, you can use the `handleSortToggle` callback function I've mentioned above.

On a side note, you're calling `sortData` here while also listening for `sortOrder` changes in your useEffect.